Maui.RevenueCat.InAppBilling/Platforms/Android/RevenueCatBillingAndroid.cs

```diff
@@ -56,48 +56,47 @@ public partial class RevenueCatBilling : IRevenueCatBilling
 
     public async partial Task<List<OfferingDto>> LoadOfferings(bool forceRefresh, CancellationToken cancellationToken)
     {
-        if (!forceRefresh && !_cachedOfferingPackages.IsNullOrEmpty())
+        if (!forceRefresh && _cachedOfferingPackages != null)
         {
-            return _cachedOfferingPackages.ToOfferDtoList(); ;
+            return _cachedOfferingPackages.ToOfferingDtoList().ToList();
         }
 
         try
         {
-            using var offerings = await Purchases.SharedInstance.GetOfferingsAsync(cancellationToken);
-            if (offerings is null)
+            _cachedOfferingPackages = await Purchases.SharedInstance.GetOfferingsAsync(cancellationToken);
+            if (_cachedOfferingPackages is null)
             {
                 return new();
             }
 
-            using var currentOffering = offerings.Current;
-            if (currentOffering is null)
-            {
-                return new();
-            }
-
-            _cachedOfferingPackages = currentOffering.AvailablePackages.ToList();
-
-            return _cachedOfferingPackages.ToOfferDtoList(); ;
+            return _cachedOfferingPackages.ToOfferingDtoList().ToList();
         }
         catch (Exception ex)
         {
-            _logger.LogError(ex, $"{nameof(LoadOfferings)} didn't suceed.");
+            _logger.LogError(ex, $"{nameof(LoadOfferings)} didn't succeed.");
             return new();
         }
     }
-    public async partial Task<PurchaseResult> PurchaseProduct(string offeringIdentifier, CancellationToken cancellationToken)
+    public async partial Task<PurchaseResult> PurchaseProduct(string offeringIdentifier, string packageIdentifier, CancellationToken cancellationToken)
```

Comment:
as I mentioned before, you can put offeringIdentifier directly to `PackageDto` and then you can send here just 1 parameter

Reply:
I get what you are saying and it will make the SDK cleaner for the consumer, however it means that inside the PurchaseProduct we would have to walk every offering and look at every package to find the correct one vs going directly to the correct offering and then to the correct package.

I could go either way on this one, if you still feel it should change let me know.

Reply:
> PurchaseProduct we would have to walk every offering and look at every package to find the correct on

Do we ? When OfferingIdentifier will be part of the PackageDto then you can still find immediately the correct offering (because PackageDto would contain it's identifier) and then in that offering find the package because you have also packageIdentifier. Or am I missing something ?

Reply:
No, you are right, I wasn't thinking it through correctly.  I will correct this.